### PR TITLE
port the cffi libffi build code

### PR DESCRIPTION
**Pull request by reaperhulk**
https://github.com/pyca/infra/blob/0db7401a572ba0e4d32bd47695922b29d256dce7/external-projects/Jenkinsfile-cffi-wheel-builder#L99-L131

### CODE REVIEW

File: .jenkins/Jenkinsfile-wheel-builder [hunk 1]
```diff
@@ -89,13 +89,39 @@ def build(version, label, imageName) {
...
+                    }
                     # Because we are doing this as root in the container, but we write to a mounted dir that is outside the container
                     # we need to make sure we set these files writable such that the jenkins user can delete them afterwards
                     mkdir -p tmpwheelhouse
                     mkdir -p wheelhouse
                     chmod -R 777 tmpwheelhouse
                     chmod -R 777 wheelhouse
 
+                    # We need libffi because pip wheel won't use wheels at all
+                    curl -#O https://mirrors.ocf.berkeley.edu/debian/pool/main/libf/libffi/libffi_\${  LIBFFI_VERSION}.orig.tar.gz
```
Comment (alex) [inline .jenkins/Jenkinsfile-wheel-builder]: the space after the `${` can't be right?

File: .jenkins/Jenkinsfile-wheel-builder [hunk 2]
```diff
@@ -89,13 +89,39 @@ def build(version, label, imageName) {
...
+                    check_sha256sum libffi_\${LIBFFI_VERSION}.orig.tar.gz \${LIBFFI_SHA256}
+                    tar zxf libffi_\${LIBFFI_VERSION}.orig.tar.gz
+                    # More root use permission nonsense
+                    chmod -R 777 libffi*
+                    pushd libffi-\${LIBFFI_VERSION}
+                    # CFLAGS needed to override the Makefile and prevent -march optimization
+                    # This flag set taken from Ubuntu 14.04's defaults. We should update it
+                    # to use -fstack-protector-strong if/when gcc 4.9+ is added to the
+                    # manylinux1 images.
+                    $linux32 ./configure CFLAGS="-g -O2 -fstack-protector --param=ssp-buffer-size=4 -  Wformat -Werror=format-security"
```
Comment (alex) [inline .jenkins/Jenkinsfile-wheel-builder]: extra space here too

File: .jenkins/Jenkinsfile-wheel-builder (hunk 1 above)
Comment (reaperhulk) [reply .jenkins/Jenkinsfile-wheel-builder]: Nope, that's a copy paste error due to going through two diff vim processes so I couldn't use the yank buffer. I accept no responsibility for my mistakes

Comment (alex) [reply .jenkins/Jenkinsfile-wheel-builder]: have you considered becoming a 🐴

**Pull request merged**